prow/pubsub/subscriber/subscriber.go

```diff
@@ -58,6 +61,90 @@ type prowCfgClient interface {
 	GetPostsubmitsStatic(identifier string) []config.Postsubmit
 }
 
+type InRepoConfigCacheGetter struct {
+	CookieFilePath string
+	CacheSize      int
+	Agent          *config.Agent
+	mu             sync.Mutex
+	GithubOptions  flagutil.GitHubOptions
+	DryRun         bool
+
+	CacheMap map[string]*config.InRepoConfigCache
+}
+
+func (irc *InRepoConfigCacheGetter) GetCache(cloneURI, host string) (*config.InRepoConfigCache, error) {
+	irc.mu.Lock()
+	defer irc.mu.Unlock()
+
+	var cache *config.InRepoConfigCache
+	var gitClientFactory git.ClientFactory
+
+	// Are we using github with InRepoConfig
+	if irc.GithubOptions.TokenPath != "" || irc.GithubOptions.AppPrivateKeyPath != "" {
+		if cache, ok := irc.CacheMap[irc.GithubOptions.Host]; ok {
+			return cache, nil
+		}
+		gitClient, err := irc.GithubOptions.GitClient(irc.DryRun)
+		if err != nil {
+			logrus.WithError(err).Fatal("Error getting Git client.")
+		}
+		gitClientFactory = git.ClientFactoryFrom(gitClient)
+
+		// Initialize cache for fetching Presubmit and Postsubmit information. If
+		// the cache cannot be initialized, exit with an error.
+		cache, err = config.NewInRepoConfigCache(
+			irc.CacheSize,
+			irc.Agent,
+			config.NewInRepoConfigGitCache(gitClientFactory))
+		// If we cannot initialize the cache, exit with an error.
+		if err != nil {
+			logrus.WithField("in-repo-config-cache-size", irc.CacheSize).WithError(err).Fatal("unable to initialize in-repo-config-cache")
+		}
+		irc.CacheMap[irc.GithubOptions.Host] = cache
+		return cache, nil
+	}
+
+	if irc.CacheMap == nil {
```

Comment:
CacheMap is referenced above as well, it needs to be initialized earlier

Reply:
Done